gap/PolygonalComplexes/modification.gd

```diff
@@ -1158,6 +1158,83 @@ DeclareOperation("SplitAllVertices", [IsPolygonalComplex]);
 DeclareOperation( "TetrahedralExtension", [IsSimplicialSurface, IsPosInt] );
 #! @EndGroup
 
+#! @BeginGroup TetrahedralReduction
+#! @Description
+#! Given a simplicial surface and a vertex of face degree 3 a new surface can
+#! be constructed by removing the faces incident to <K>vertex</K> in
+#! <K>surface</K> and attaching a new face to the resulting boundary edges.
+#! <Alt Only="TikZ">
+#!      \input{Image_TetraRed.tex}
+#! </Alt>
+#! So, this modification results in decreasing the number of vertices by 1, the 
+#! number of edges by 3 and the number of faces by 2. As an Example consider
+#! the double tetrahedron.
+#! <Alt Only="TikZ">
+#!      \input{Image_DoubleTetrahedron.tex}
+#! </Alt>
+#! @BeginExampleSession
+#! gap> vof:=[[1,2,4],[2,3,4],[1,3,4],[1,2,5],[1,3,5],[2,3,5]];;
+#! gap> doubleTetra:=SimplicialSurfaceByVerticesInFaces(vof);
+#! simplicial surface (5 vertices, 9 edges, and 6 faces)
+#! gap> tet:=TetrahedralReduction(doubleTetra,4);
+#! simplicial surface (4 vertices, 6 edges, and 4 faces)
+#! gap> IsIsomorphic(tet,Tetrahedron());
+#! true
+#! @EndExampleSession
+
+#! So performing the tetrahedral reduction with the double tetrahedron gives rise
+#! to the tetrahedron.
+#! @Returns a surface
+#! @Arguments surface,vertex 
+DeclareOperation( "TetrahedralReduction", [IsSimplicialSurface,IsPosInt] );
+#! @EndGroup
+
+#! @BeginGroup InnerMultiTetrahedralSphere
+#! @Description
+#! Return the inner multi tetrahedral sphere of a given twisted polygonal
+#! complex. If <K>complex</K> is a multi tetrahedral sphere, a new multi
+#! tetrahedral sphere can be obtained by applying a tetrahedral reduction
+#! to every vertex in <K>complex</K> with face degree 3. If <K>complex</K>
+#! is not a multi tetrahedral sphere, the function returns <K>fail</K>.
+#!
+#! As an example, consider the multi tetrahedral sphere obtained by applying 
+#! exactly two tetrahdral extensions to the tetrahedron with any two faces.
+#! Since this surface has exactly two vertices with face degree 3, removing 
+#! the attached tetrahedron gives rise to the tetrahedron.
+#! @BeginExampleSession
+#! gap> doubleTetra:=TetrahedralExtension(Tetrahedron(),1);
+#! simplicial surface (5 vertices, 9 edges, and 6 faces)
+#! gap> multiTetra:=TetrahedralExtension(doubleTetra,2);
+#! simplicial surface (6 vertices, 12 edges, and 8 faces)
+#! gap> tetra:=InnerMultiTetrahedralSphere(multiTetra);
+#! simplicial surface (4 vertices, 6 edges, and 4 faces)
+#! gap> IsIsomorphic(tetra,Tetrahedron());
+#! true
+#! @EndExampleSession
+#! @Returns a complex
+#! @Arguments complex 
+DeclareOperation( "InnerMultiTetrahedralSphere", [IsTwistedPolygonalComplex] );
+## ! @EndGroup 
+
+#! @BeginGroup MultiTetrahedralSphereByTetrahedralSymbol
+#! @Description
+#! Returns a multi tetrahedral sphere with the given multi tetrahedral symbol.
```

Comment:
Where is the tetrahedral symbol defined?

Reply:
I wanted to add a method "TetrahedralSymbolOfComplex" to the package, where I am going to describe the symbol and then I am going to use a reference in this part.

Reply:
That's good! But without a definition it's difficult for me to unterstand the implementation.